### DISCUSSION TITLE
Auto-reloading sandwich TUI workflow?

### DISCUSSION THREAD

**Discussion by ryantm** [codedownio/sandwich#33] [Q&A] [answered]
Hi.

It would be nice to reload my sandwich TUI when either my tests or code change. Does anyone have a good solution for this?

I tried:

```ShellSession
$ find *.hs | entr -r bash -c 'cabal v2-run test -- --tui'
Up to date
Some formatters failed: '[getTerminalAttributes: illegal operation (Inappropriate ioctl for device)]'
1 runs succeeded out of 1 repeats
```

It works with normal entr if I manually exit out of the tui after editing the files.

**Comment by thomasjm** [comment]
Hmm, I notice that `find -name "*.hs" | entr -r htop` doesn't work either. I suspect `entr` isn't hooking the subprocess up to a proper pty.

> It works with normal entr if I manually exit out of the tui after editing the files.

Could you explain how you got that to work?

I've daydreamed about making a Sandwich reloader that runs a GHCi subprocess like how `ghcid` does, and magically reloads code when files change...

**Reply by thomasjm** [reply]
Oh, I see that it launches TUI apps successfully without the `-r` flag. This does seem like a problem with `entr`.

**Reply by thomasjm** [reply]
Filed https://github.com/eradman/entr/issues/81

**Reply by thomasjm** [reply, marked as answer]
BTW, I should mention another solution that came up on #98:

https://github.com/codedownio/sandwich/issues/98#issuecomment-2442988029

I think this will be the "answer" for now, since the amount of engineering effort to do this without something like `watchexec` is still prohibitive.
